pkg/controller/utils/utils.go

```diff
@@ -350,3 +350,38 @@ func BuildControllerLogger(controller hivev1.ControllerName, resource string, ns
 		"reconcileID": utilrand.String(constants.ReconcileIDLen),
 	})
 }
+
+// SetProxyEnvVars will add the standard proxy environment variables to all containers in the given pod spec. If
+// any of the provided values are empty, the environment variable will not be set.
+func SetProxyEnvVars(podSpec *corev1.PodSpec, httpProxy, httpsProxy, noProxy string) {
+	setEnvVarOnContainers := func(podSpec *corev1.PodSpec, envVar, value string) {
+		if value != "" {
+			for i := range podSpec.Containers {
+				// Check if the env var is already set, if so we preserve the original but warn if they differ:
```

Comment:
this sounds off from the goal of the SetProxyEnvVars .. i would assume we force our value.